Microsoft.Toolkit.Uwp.Services/Services/MicrosoftTranslator/TranslatorService.cs

```diff
@@ -87,13 +93,55 @@ public string SubscriptionKey
         public string Language { get; set; }
 
         /// <summary>
+        /// Detects the language of a text.
+        /// </summary>
+        /// <param name="text">A string represeting the text whose language must be detected.</param>
+        /// <returns>A string containing a two-character Language code for the given text.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <list type="bullet">
+        /// <term>The <see cref="SubscriptionKey"/> property hasn't been set.</term>
+        /// <term>The <paramref name="text"/> parameter is <strong>null</strong> (<strong>Nothing</strong> in Visual Basic) or empty.</term>
+        /// </list>
+        /// </exception>
+        /// <exception cref="TranslatorServiceException">The provided <see cref="SubscriptionKey"/> isn't valid or has expired.</exception>
+        /// <remarks><para>This method performs a non-blocking request for language detection.</para>
+        /// <para>For more information, go to https://docs.microsofttranslator.com/text-translate.html#!/default/get_Detect.
+        /// </para></remarks>
+        /// <seealso cref="GetLanguagesAsync"/>
+        public async Task<string> DetectLanguageAsync(string text)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            text = text.Substring(0, Math.Min(text.Length, _MaxTextLengthForAutoDetection));
+
+            // Checks if it is necessary to obtain/update access token.
+            await CheckUpdateTokenAsync().ConfigureAwait(false);
+
+            var uriString = string.Format(DetectUri, Uri.EscapeDataString(text));
+
+            using (var request = CreateHttpRequest($"{BaseUrl}{uriString}"))
+            {
+                var response = await HttpHelper.Instance.SendRequestAsync(request).ConfigureAwait(false);
+                var content = await response.GetTextResultAsync().ConfigureAwait(false);
+
+                var doc = XDocument.Parse(content);
+                var detectedLanguage = doc.Root.Value;
+
+                return detectedLanguage;
```

Comment:
Yes, it is surely possible, but another call to the service will be necessary for each detection, unless I keep a local cache of all the Language Names. Otherwise, you need to manually call GetLanguageNamesAsync method and then keep the ServiceLanguage list in our app, in order to retrive the full name corresponding to the detected language.